tools/common/Tuning.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.IO;
+using System.Text;
+
+using Mono.Linker;
+using Mono.Linker.Steps;
+
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+using Xamarin.Bundler;
+
+#if MONOTOUCH
+using PlatformException = Xamarin.Bundler.MonoTouchException;
+#else
+using PlatformException = Xamarin.Bundler.MonoMacException;
+#endif
+
+#if MONOTOUCH
+namespace MonoTouch.Tuner {
+#else 
+namespace MonoMac.Tuner {
+#endif
+	static partial class Linker {
+		static void HandlePipelineProcessException (Exception e)
```

Comment:
Good idea. Let me try that.

Reply:
So it turns out I did that so I could recursively call it here:

https://github.com/xamarin/xamarin-macios/pull/7628/files#diff-8254938952b7cbaad15f08f2e26cb8b4R59

and dig out the InnerException. Thoughts?

Reply:
hmm... go back to original but don't throw, just return the exception to br thrown
and if the exception you return is identical to the original one then rethrow (instead of throwing)

```csharp
static Exception HandlePipelineProcessException (Exception e) { ... }

...
catch (Exception e) {
   var pe = HandlePipelineProcessException (e);
   if (e == pe)
      rethrow;
   throw pe;
}
```